rewrite-java/src/main/java/org/openrewrite/java/cleanup/UnnecessaryParenthesesVisitor.java

```diff
@@ -223,4 +216,27 @@ public J visitUnary(J.Unary unary, P ctx) {
         }
         return u;
     }
+
+    @Override
+    public J visitMethodInvocation(J.MethodInvocation method, P p) {
+        J.MethodInvocation mi = (J.MethodInvocation) super.visitMethodInvocation(method, p);
+        if (mi.getSelect() instanceof J.Parentheses &&
+                !(((J.Parentheses<?>) mi.getSelect()).getTree() instanceof J.Binary)) {
```

Comment:
I wonder if we also need to check for some other cases like J.Ternary and J.TypeCast.

Reply:
Thanks! Addressed in https://github.com/openrewrite/rewrite/commit/cc4e2c063c56a25bab3118cede3373ca7b3a03d5